crates/typed-store-derive/src/lib.rs

```diff
@@ -72,10 +78,24 @@ fn extract_struct_info(
             let type_str = format!("{}", &type_info.ident);
             // Rough way to check that this is map_type_name
             if allowed_map_type_names.contains(&type_str) {
-                return (
-                    (f.ident.as_ref().unwrap().clone(), type_str),
-                    (inner_type, options),
-                );
+                let field_name = f.ident.as_ref().unwrap().clone();
+                let path_name = if let Some(rename) = attrs.get(DB_OPTIONS_RENAME) {
```

Comment:
minor personal nit: but maybe `cf_name` is a better name for this (vs `path_name`)